Update PPP (#250)
AY2223S2-CS2103-W17-2/tp: docs/team/valerietanhx.md

### Project: Dengue Hotspot Tracker

Dengue Hotspot Tracker is a desktop app for managing dengue cases.
The user interacts with it using a CLI, and it has a GUI created with JavaFX.
It is written in Java, and has about 9.5 kLoC.

Given below are my contributions to the project.

* **New Feature:** Added the ability to delete multiple indexes at once.
  * Justification: This feature allows users to delete cases in batches, rather than having to do so one at a time.
* **New Feature:** Added the ability to delete all cases from a given date.
  * Justification: This feature allows users to delete all cases from a given date at once, which may be useful

>     when the user wants to clear older or erroneous cases from the same date from the DengueHotspotTracker.

Reviewer: ```suggestion
    when the user wants to clear older or erroneous cases from the same date from the `DengueHotspotTracker`.
```

>   * Highlights: This enhancement was challenging as it required significant changes to the original delete command parser.

Author: ```suggestion
  * Highlights: This enhancement was challenging as it required significant changes to the original `delete` command parser.
```

>     It caused the delete command to be the only command that could accept indexes and prefixes, _but never together_.

Reviewer: ```suggestion
    It caused the `delete` command to be the only command that could accept indexes and prefixes, _but never together_.
```

    This meant that much effort had to be put into ensuring that the parser could accurately detect when the two had been mixed,
    regardless of the index(es) came before or after the date, and throw exceptions accordingly.

>     The delete command itself was also updated to use Optionals instead, as it now had to hold both indexes and dates,

Reviewer: ```suggestion
    The `delete` command itself was also updated to use the `Optional` class instead, as it now had to hold both indexes and dates,
```

    but never at the same time. Much abstraction was required to keep the code clean and readable.
* **New Feature:** Added the ability to delete all cases from a given date range.
  * Justification: This feature allows users to delete all cases from a given date at once, which may be useful

>     when the user wants to clear erroneous cases from the same date from the DengueHotspotTracker.

Reviewer: ```suggestion
    when the user wants to clear erroneous cases from the same date from the `DengueHotspotTracker`.
```

  * Highlights: Similar to the previous feature, this enhancement was challenging as it required significant changes

>     to the architecture of the delete command parser and delete command. In addition to the aforementioned changes

Reviewer: ```suggestion
    to the architecture of the `delete` command parser and `delete` command. In addition to the aforementioned changes
```

>     to the architecture of the delete command parser and delete command. In addition to the aforementioned changes

Reviewer: ```suggestion
    to the architecture of the `delete` command parser and `delete` command. In addition to the aforementioned changes
```

>     to the parser and use of Optionals, new classes such as `Range`, `StartDate`, and `EndDate`, as well as

Reviewer: ```suggestion
    to the parser and use of the `Optional` class, new classes such as `Range`, `StartDate`, and `EndDate`, as well as
```

>     to the parser and use of Optionals, new classes such as `Range`, `StartDate`, and `EndDate`, as well as

Reviewer: ```suggestion
    to the parser and use of the `Optional` class, new classes such as `Range`, `StartDate`, and `EndDate`, as well as
```
